src/Sulu/Bundle/TrashBundle/Infrastructure/Symfony/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TrashBundle\Infrastructure\Symfony\DependencyInjection;
+
+use Sulu\Bundle\TrashBundle\Domain\Model\TrashItem;
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder('sulu_trash');
+        $rootNode = $treeBuilder->getRootNode();
+        $rootNode->children()
+            ->arrayNode('restore_form')
+                ->useAttributeAsKey('resourceKey')
+                ->scalarPrototype()
+            ->end()
+        ->end();
+
+        $this->addObjectsSection($rootNode);
+
+        return $treeBuilder;
+    }
+
+    private function addObjectsSection(ArrayNodeDefinition $node): void
+    {
+        $node->children()
+            ->arrayNode('objects')
+                ->addDefaultsIfNotSet()
+                ->children()
+                    ->arrayNode('trash_item')
+                        ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('model')->defaultValue(TrashItem::class)->end()
```

Comment:
As written in #5931, our repositories should be more than just doctrine repositories. Therefore they should also not extend from `EntityRepository`. But when configuring the `repository` here, the `TrashItemRepository` **is** the doctrine repository. I guess it would be better to instead inject the doctrine repository into the `TrashItemRepository`, because both of them are and should be different services. If someone wants to override the `TrashItemRepository`, he can just decorate the `TrashItemRepository` service. Also, `$em->getRepository(TrashItem::class)` should return the doctrine repository instead of the `TrashItemRepository`, because the latter one isn't an `EntityRepository` and I would expect `$em->getRepository()` to return an `EntityRepository`

Reply:
sounds reasonable 🙂  i guess it would be better to use existing symfony mechanisms (eg. decoration) instead of implementing a custom mechanism in the `PersistenceBundle` for overwriting repositories in the future